treatment-service/models/storage.go

```diff
@@ -238,6 +234,22 @@ func (i *ExperimentIndex) isActive() bool {
 	return (i.StartTime.Before(time.Now()) || i.StartTime.Equal(time.Now())) && i.EndTime.After(time.Now())
 }
 
+func (i *ExperimentIndex) checkSegmentHasWeakMatch(segmentName string) bool {
+	if set, exists := i.stringSets[segmentName]; !exists || set.Len() == 0 {
+		return true
+	}
+	if set, exists := i.intSets[segmentName]; !exists || set.Len() == 0 {
+		return true
+	}
+	if set, exists := i.realSets[segmentName]; !exists || set.Len() == 0 {
+		return true
+	}
+	if _, exists := i.boolFlags[segmentName]; !exists {
+		return true
+	}
+	return false
+}
+
```

Comment:
This new function is basically performing the same purpose as 

```go
if _, exists := i.Experiment.Segments[segmentName]; !exists ||
			len(i.Experiment.Segments[segmentName].GetValues()) == 0 {
// ...
}
```
but has been refactored into a separate helper function and also uses the values within the sets of `ExperimentIndex` instead of the values stored in `ExperimentIndex.Experiment.Segments` (which would have been deleted).